docs/_pages/typecheck.md

```diff
@@ -399,3 +399,63 @@ return module
 ```
 
 There are some caveats here though. For instance, the require path must be resolvable statically, otherwise Luau cannot accurately type check it.
+
+## Cyclic modules
+
+Luau will warn about cycles in module requirements, for example if `A.lua` is:
+
+```lua
+--!strict
+local B = require(script.Parent.B)
+local module = {}
+module.x = 5
+module.z = B.y
+return module
+```
+
+and `B.lua` is:
+
+```lua
+--!strict
+local A = require(script.Parent.A)
+local module = {}
+module.y = 7
+module.z = A.x
+return module
+```
+
+then there are errors:
+
+```
+A.lua:2:11-34: (E001) Cyclic module dependency detected at require() call.
+B.lua:2:11-34: (E001) Cyclic module dependency detected at require() call.
+```
+
+Most of the time, this is what you want, since at runtime cyclic
+modules never finish loading, but in some cases, there may be runtime
+checks to break cycles, such as `require`ing a module inside a
+function.  For example we could break the cycle between `A` and `B` by
+changing `B.lua`:
+
+```lua
+--!strict
+local module = {}
+local function loadA() return require(script.Parent.A) end
+module.x = 5
+function module.getZ() return loadA().x end
+return module
+```
+
+Now the cycle is broken at runtime, but there is still a cycle in the
+types: in order to know the type of `A`, we need to the type of `B`,
+for which we need the type of `A`.
+
+In order to break the cycle of types, we mark one of the `require`s as having
+type `any`, for example:
```

Comment:
I wonder if we need to be precise here and talk about marking the `require` argument, since 'marking the require' could be thought as ```require(script.Parent.B) :: any``` (which actually works right now, but is fragile wrt pattern matching in the code)

Reply:
I reworded this as "... mark one of the modules as having ...", does that capture the meaning?